src/PowerShellEditorServices/Extensions/Api/EditorUIService.cs

```diff
@@ -0,0 +1,171 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Services.PowerShellContext;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PowerShellEditorServices.Extensions.Api
+{
+    /// <summary>
+    /// Object specifying a UI prompt option to display to the user.
+    /// </summary>
+    public sealed class PromptChoiceDetails
+    {
+        /// <summary>
+        /// Construct a prompt choice object for display in a prompt to the user.
+        /// </summary>
+        /// <param name="label">The label to identify this prompt choice. May not contain commas (',').</param>
+        /// <param name="helpMessage">The message to display to users.</param>
+        public PromptChoiceDetails(string label, string helpMessage)
+        {
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            if (label.Contains(","))
+            {
+                throw new ArgumentException($"Labels may not contain ','. Label: '{label}'", nameof(label));
+            }
```

Comment:
Ideally this would change somehow, even if just for single choice prompts.  It may not seem likely to have <kbd>,</kbd>, but I have one where you select from available method overloads to decorate arguments with `<# parameterName: #>`.  To get that working, for right now I'm just replacing <kbd>,</kbd> with <kbd>0xFF0C</kbd> which looks real weird but is functional.

Reply:
Yeah this is another spot where the client has some bad logic in it.

Rather than expose its current flaws in the api, I’ve limited its surface for now until the client is fixed.

The client currently sends back labels concatenated with commas, hence why they’re not allowed here. Once we change the message format we can fix it